### ISSUE TITLE
Asynchronous component update not working

### ISSUE THREAD

**Issue by marians** [giantswarm/cli#472]
I compiled the CLI from current master (`7e19cd9d57`) and tried to update a component on `aws-alpha` by specifying a certain image tag to use. Before the component used `:latest`, afterwards it should use `:0.0.1` (which actually points to the same image).

State before:

```
$ swarm status
Service 'eventtracking' is up

component  image                                         instanceid    created              status
couchdb    reg…/giantswarm/eventtracking-couchdb:latest  g2pe0odtz48c  2015-11-05 09:21:28  up
flask      reg…/giantswarm/eventtracking-flask:latest    e5g4vu91wbz3  2015-11-05 11:18:00  up
flask      reg…/giantswarm/eventtracking-flask:latest    ncmhfiotxvbt  2015-11-05 11:18:02  up
flask      reg…/giantswarm/eventtracking-flask:latest    rpwz54f28izu  2015-11-05 11:00:26  up
```

Output:

```
$ swarm update eventtracking/flask 0.0.1
Updating component 'eventtracking/flask' with version '0.0.1' and strategy 'one-by-one'…
\ 
```

State after:

```
Service 'eventtracking' is down

component  image                                                           imageid       instanceid    podid         created              status
couchdb    registry.giantswarm.io/giantswarm/eventtracking-couchdb:latest  4e3ff2b61262  g2pe0odtz48c  g2pe0odtz48c  2015-11-05 09:21:28  up
flask      registry.giantswarm.io/giantswarm/eventtracking-flask:0.0.1     492890165fda  e5g4vu91wbz3  e5g4vu91wbz3  2015-11-05 11:18:00  up
flask      registry.giantswarm.io/giantswarm/eventtracking-flask:0.0.1     492890165fda  ncmhfiotxvbt  ncmhfiotxvbt  2015-11-05 11:18:02  up
flask      registry.giantswarm.io/giantswarm/eventtracking-flask:0.0.1     n/a           rpwz54f28izu  rpwz54f28izu  2015-11-05 11:00:26  down
```

State hasn't changed for a while now.

Ping @zyndiecate @ewoutp

**Comment by marians** [comment]
Seems like the Task ID for the component update was `i2af9tmwwrh4til0`

**Comment by zyndiecate** [comment]
Can you please provide the definition file and org/env?

**Comment by marians** [comment]
Env: `giantswarm/testing`
Service definition:
```
{
  "name": "eventtracking",
  "components": {
    "flask": {
      "image": "registry.giantswarm.io/giantswarm/eventtracking-flask:latest",
      "ports": [5000],
      "links": [
        {
          "component": "couchdb",
          "target_port": 5984
        }
      ],
      "env": {
        "COUCHDB_PASS": "..."
      },
      "domains": {
        "5000": "..."
      }
    },
    "couchdb": {
      "image": "registry.giantswarm.io/giantswarm/eventtracking-couchdb:latest",
      "ports": [5984],
      "env": {
        "COUCHDB_PASS": "..."
      },
      "domains": {
        "5984": "..."
      },
      "volumes": [{
        "path": "/usr/local/var/lib/couchdb",
        "size": "6 GB"
      }]
    }
  }
}
```

**Comment by zyndiecate** [comment]
Your status example shows that one component is scaled 3 times. How did you do that? Scale one time to 3 or 3 times to 1 each?

**Comment by zyndiecate** [comment]
I checked that with Marian. 
- AWS alpha seems to be broken at the moment.
- AWS cluster-01 worked with this setup.
- We scaled to 6 flask instances and the update broke. 
- Instances were navigated into failed state. Once this state is reached the user is not able anymore to fix it. I restarted fleet units, besides one unit chain, which was completely gone. 

See the following information we found.

```
Error delete Instance: retry limit reached (10/10): googleapi: Error 409: unit does not exist and options field empty | {\"task-description\":\"Delete pod 'qk85tduxat13' from scheduler\",\"task-id\":\"2fxf27cme69wudxd\",\"task-worker-id\":\"cq83c4mbrxtw3j4n\"}
```

```
$ SWARM_CLUSTER_ID=aws-cluster-01.giantswarm.io swarm --env giantswarm/testing logs qk85tduxat13
2015-11-06 10:03:16.526 +0000 UTC    - docker  - Creating database 'events'
2015-11-06 10:03:16.582 +0000 UTC    - docker  - * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
2015-11-06 10:05:48.732 +0000 UTC    - docker  - Repository registry.giantswarm.io/giantswarm/eventtracking-flask already being pulled by another client. Waiting.
2015-11-06 10:05:56.472 +0000 UTC    - docker  - * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
2015-11-06 10:09:45.605 +0000 UTC    - docker  - Repository registry.giantswarm.io/giantswarm/eventtracking-flask already being pulled by another client. Waiting.
2015-11-06 10:09:54.025 +0000 UTC    - docker  - * Running on http://0.0.0.0:5000/ (Press CTRL+C to quit)
```

```
INFO reconcile.go:330: AgentReconciler completed task: type=UnloadUnit job=ig-00000000-qk85tduxat13-006-qk85tduxat13-lb-01.service reason=\"unit loaded but not scheduled here
```

```
INFO manager.go:261: Removing systemd unit ig-00000000-qk85tduxat13-003-qk85tduxat13-ne-01.service
```

Here is the kibana link: http://kibana.monitoring.giantswarm.io/#/discover?_g=(time:(from:'2015-11-06T10:44:24.702Z',mode:absolute,to:'2015-11-06T10:52:30.000Z'))&_a=(columns:!(_source),index:'logstash-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'%2Blogtext:%22qk85tduxat13%22%20-logtext:%22skipped%22%20-logtext:%22Assuming%20failed%22%20-logtext:%22%3D%3E%20failed%22%20-logtext:%22logs%3Ff%22')),sort:!('@timestamp',desc))


@ewoutp @htr any ideas?

**Comment by zyndiecate** [comment]
@marians as investigated and fixed in https://github.com/giantswarm/cli/pull/474, the update itself works. Problem was an unhandled gateway timeout error. I have no idea which component in our micoservice architecture throws this, but the PR fixes it. So handling this HTTP 504 causes the command not to break after 4 minutes. 

Further I played around with component updates as you described with the exact same service on cluster-01. I updated back and forth, at least 10 times. All works well. Looks like the other issues you experienced was caused by general unstability  of our underlying systems.
